# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/worklet/WorkletFetchHandler.cpp
# Commit: cc6bc73fbbfd
# Full Hash: cc6bc73fbbfd2dd796ae51f2b85c7e9c24ae2686
# Author: Alexandru Marc <amarc@mozilla.com>
# Date: 2025-09-25 05:12 +0300
# Regressor Bug: 1968890
# File Overlap Count: 14
# Overlapping Files: dom/base/test/jsmodules/chrome.toml, dom/base/test/jsmodules/test_dynamicImportErrorMessage.html, dom/script/ScriptLoadContext.cpp, dom/script/ModuleLoader.cpp, dom/base/test/jsmodules/import_404.mjs
# Description:
#   Revert "Bug 1968890 - Part 9: Add test for dynamic import. r=jonco" for causing multiple wpt failures
# ==============================================================================

diff -r e1c5be04b3a0 -r cc6bc73fbbfd dom/worklet/WorkletFetchHandler.cpp
--- a/dom/worklet/WorkletFetchHandler.cpp	Wed Sep 24 23:17:54 2025 +0000
+++ b/dom/worklet/WorkletFetchHandler.cpp	Thu Sep 25 05:12:32 2025 +0300
@@ -220,6 +220,15 @@
 
   request->mBaseURL = mURI;
   request->OnFetchComplete(mResult);
+
+  if (NS_FAILED(mResult)) {
+    if (request->IsTopLevel()) {
+      request->LoadFailed();
+    } else {
+      request->Cancel();
+    }
+  }
+
   moduleLoader->RemoveRequest(mURI);
   return NS_OK;
 }